### PR TITLE
Use kubeadm stable for systesting

### PR DESCRIPTION
**Pull request by tiewei**
This commit updated to use kubeadm stable for system testing, and
created a single base yaml file for contiv deployment, also created
a contiv-compose to update the base yaml file to support system testing,
prometheus, auth_proxy and ACI(not yet).
Update system testings to allow running netplugin on master nodes, so that
service like kube-dns and kube-dashboard could be created correctly.
Also formated ruby files and updated vagrant file so kube master could have
more memories.

Signed-off-by: Wei Tie <wtie@cisco.com>

### CODE REVIEW

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

File: install/k8s/cluster/bootstrap_centos.sh [hunk 1]
```diff
@@ -1,20 +1,42 @@
...
+
 set -e
+
+swapoff -a
+setenforce 0
+systemctl stop firewalld
+systemctl disable firewalld
+/etc/init.d/network restart
+
+cp /shared/.etc_hosts /etc/hosts
```
Comment (chrisplo) [inline install/k8s/cluster/bootstrap_centos.sh]: seems bad to have this as a hidden file?

Comment (tiewei) [reply install/k8s/cluster/bootstrap_centos.sh]: it was so, do you want me to change ?

File: install/k8s/contiv/contiv-compose [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+                            'name'] not in ('CONTIV_NETPLUGIN_ETCD_ENDPOINTS',
+                                            'CONTIV_NETMASTER_ETCD_ENDPOINTS')
+                    ]
+                    # make the container idle
+                    container['command'] = ["tail", "-f", "/dev/null"]
+                    # add binary mount points to run tests
+                    container['volumeMounts'].append({
+                        'mountPath': '/contiv/bin',
+                        'name': 'contiv-bin-dir',
+                        'readOnly': False
```
Comment (chrisplo) [inline install/k8s/contiv/contiv-compose]: why not true?

File: install/k8s/contiv/contiv-compose [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+                    })
+                    container['volumeMounts'].append({
+                        'mountPath': '/contiv/scripts/',
+                        'name': 'contiv-scripts-dir',
+                        'readOnly': False
```
Comment (chrisplo) [inline install/k8s/contiv/contiv-compose]: why not true?

File: install/k8s/contiv/contiv-compose [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+    def _is_resource(self, resource, kind, name, namespace=None):
+        return (resource['kind'] == kind and
+                resource['metadata']['name'] == name and
+                resource['metadata'].get('namespace') == namespace)
+
+    def _update_images(self, resource, args):
+        if self._is_resource(resource, 'DaemonSet', 'contiv-netplugin',
+                             'kube-system') or self._is_resource(
+                                 resource, 'ReplicaSet', 'contiv-netmaster',
+                                 'kube-system'):
```
Comment (chrisplo) [inline install/k8s/contiv/contiv-compose]: helper function?  copy/pasta

File: install/k8s/cluster/bootstrap_centos.sh [hunk 5]
```diff
@@ -38,14 +38,16 @@ EOF
...
 sysctl --system
 
-yum remove -y docker \
-                  docker-common \
-                  container-selinux \
-                  docker-selinux \
-                  docker-engine \
-                  docker-engine-selinux
+yum remove -y docker docker-common container-selinux docker-selinux docker-engine docker-engine-selinux
+yum install -y docker
```
Comment (chrisplo) [inline install/k8s/cluster/bootstrap_centos.sh]: curious why docker was removed above just to be reinstalled, is there something in pre/post installation scripts?

File: install/k8s/contiv/base.yaml [hunk 6]
```diff
@@ -79,6 +77,7 @@ rules:
       - namespaces
       - networkpolicies
     verbs:
+      - get
```
Comment (chrisplo) [inline install/k8s/contiv/base.yaml]: does this fix all the woes?

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

**Comment by vhosakot** [comment]
Per our discussion, the files `prometheus.yml`, `contiv-grafana.yml` and `contiv-prometheus.yml` must be migrated from https://github.com/contiv/install/tree/master/install/k8s/configs to `contiv/netplugin` so that they can be used as working sample/reference by CCP devs when CCP stands up Prometheus and Grafana after EAP.

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

File: install/k8s/contiv/contiv-compose (hunk 2 above)
Comment (tiewei) [reply install/k8s/contiv/contiv-compose]: need execute

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

**Comment by tiewei** [comment]
build PR

File: Makefile [hunk 7]
```diff
@@ -135,7 +136,8 @@ start:
 # ===================================================================
 # kubernetes cluster bringup/cleanup targets
 k8s-cluster:
-	cd vagrant/k8s/ && CONTIV_K8S_USE_KUBEADM=1 ./setup_cluster.sh
+	vagrant plugin install vagrant-cachier
```
Comment (tiewei) [inline Makefile]: if it failed, should still go to next install

File: Makefile [hunk 8]
```diff
@@ -147,9 +149,14 @@ k8s-l3-destroy:
...
+# kubernetes dev
+k8s-dev:
+	CONTIV_TEST=1 make k8s-cluster
+	cd vagrant/k8s/ && vagrant ssh k8master -c 'bash -lc "cd /opt/gopath/src/github.com/contiv/netplugin && make deps checks clean compile"'
+
 # kubernetes test targets
-k8s-test: k8s-cluster
-	cd vagrant/k8s/ && vagrant ssh k8master -c 'bash -lc "cd /opt/gopath/src/github.com/contiv/netplugin && make run-build"'
+k8s-test:
+	make k8s-dev
```
Comment (tiewei) [inline Makefile]: make `k8s-dev` as dependency

File: scripts/netContain/scripts/contivNet.sh [hunk 9]
```diff
@@ -50,53 +50,14 @@ echo "INFO: Running contiv in mode $CONTIV_MODE"
...
-    if [ "$CONTIV_ROLE" = "netplugin" ]; then
-        mkdir -p /opt/cni/bin
-        cp /contiv/bin/contivk8s /opt/cni/bin/
-        mkdir -p /etc/cni/net.d/
-        set -x
-        echo ${CONTIV_CNI_CONFIG} > /etc/cni/net.d/1-contiv.conf
-        set +x
-    fi
-fi
+mkdir -p /opt/contiv/
```
Comment (tiewei) [inline scripts/netContain/scripts/contivNet.sh]: not needed

File: install/k8s/cluster/bootstrap_centos.sh [hunk 10]
```diff
@@ -23,15 +45,7 @@ yum remove -y docker \
                   docker-engine \
                   docker-engine-selinux
 
-yum install -y docker ebtables \
-	https://fedorapeople.org/groups/kolla/kubeadm-1.6.0-0.alpha.0.2074.a092d8e0f95f52.x86_64.rpm \
-	https://fedorapeople.org/groups/kolla/kubectl-1.5.4-0.x86_64.rpm \
-	https://fedorapeople.org/groups/kolla/kubelet-1.5.4-0.x86_64.rpm \
-	https://fedorapeople.org/groups/kolla/kubernetes-cni-0.3.0.1-0.07a8a2.x86_64.rpm
+yum install -y docker kubelet kubeadm kubectl
```
[the remark this answers is not in the document]
Comment (chrisplo) [reply install/k8s/cluster/bootstrap_centos.sh]: version support added

**Comment by tiewei** [comment]
Build PR

**Comment by chrisplo** [comment]
thanks for the minor feedback patches

**Pull request merged**